File: backend/src/modules/transportation/costs/costs.controller.ts
import { Controller, Get, Headers, UseGuards } from '@nestjs/common';
import { HeaderAuthGuard } from '../../../common/auth/header-auth.guard';
import { CostsService } from './costs.service';

// Base path /api/Transportation (faithful to the documented URLs, §6.11); every
// action requires the CompanyName + UserToken headers via the guard. Filters and
// pagination ride in HTTP headers (Node lowercases them) — exactly as documented.
@Controller('api/Transportation')
@UseGuards(HeaderAuthGuard)
export class CostsController {
  constructor(private readonly service: CostsService) {}

  @Get('ReportCostOfLines')
  reportCostOfLines(
    @Headers('pageno') pageNo?: string,
    @Headers('noofitems') noOfItems?: string,
    @Headers('supplierid') supplierId?: string,
    @Headers('transportionlineid') lineId?: string,
    @Headers('serialbus') serialBus?: string,
    @Headers('datefrom') dateFrom?: string,
    @Headers('dateto') dateTo?: string,
  ) {
    return this.service.reportCostOfLines(Number(pageNo) || 1, Number(noOfItems) || 20, {
      supplierId: Number(supplierId) || undefined,
      lineId: Number(lineId) || undefined,
      serialBus: serialBus || undefined,
      dateFrom: dateFrom || undefined,
      dateTo: dateTo || undefined,
    });
  }

  @Get('ReportCostOfLinesExcell')
  reportCostOfLinesExcel(
    @Headers('supplierid') supplierId?: string,
    @Headers('transportionlineid') lineId?: string,
    @Headers('serialbus') serialBus?: string,
    @Headers('datefrom') dateFrom?: string,
    @Headers('dateto') dateTo?: string,
  ) {
    return this.service.reportCostOfLinesExcel({
      supplierId: Number(supplierId) || undefined,
      lineId: Number(lineId) || undefined,
      serialBus: serialBus || undefined,
      dateFrom: dateFrom || undefined,
      dateTo: dateTo || undefined,
    });
  }
}

File: backend-dotnet/GarasTransport.Api/Modules/Transportation/Costs/CostsController.cs
using GarasTransport.Api.Common;
using Microsoft.AspNetCore.Mvc;

namespace GarasTransport.Api.Modules.Transportation.Costs;

// Base path /api/Transportation (§6.11). Every action requires CompanyName +
// UserToken headers via the filter.
[Route("api/Transportation")]
[HeaderAuth]
public class CostsController : ApiControllerBase
{
    private readonly CostsService _service;

    public CostsController(CostsService service) => _service = service;

    private CostFilters Filters() => new()
    {
        SupplierId = HeaderIntOrNull("SupplierId"),
        LineId = HeaderIntOrNull("TransportionLineId"),
        SerialBus = Header("SerialBus"),
        DateFrom = Header("DateFrom"),
        DateTo = Header("DateTo"),
    };

    [HttpGet("ReportCostOfLines")]
    public Task<object> ReportCostOfLines() =>
        _service.ReportCostOfLines(HeaderInt("PageNo", 1), HeaderInt("NoOfItems", 20), Filters());

    [HttpGet("ReportCostOfLinesExcell")]
    public Task<object> ReportCostOfLinesExcel() => _service.ReportCostOfLinesExcel(Filters());
}
